src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_UsingStatement.cs

```diff
@@ -292,7 +292,11 @@ private BoundStatement RewriteUsingStatementTryFinally(SyntaxNode syntax, BoundB
             BoundExpression disposeCall;
 
             MethodSymbol disposeMethodSymbol;
-            if (Binder.TryGetSpecialTypeMember(_compilation, SpecialMember.System_IDisposable__Dispose, syntax, _diagnostics, out disposeMethodSymbol))
+            if (!(methodOpt is null))
+            {
+                disposeCall = BoundCall.Synthesized(syntax, disposedExpression, methodOpt);
+            }
+            else if (Binder.TryGetSpecialTypeMember(_compilation, SpecialMember.System_IDisposable__Dispose, syntax, _diagnostics, out disposeMethodSymbol))
```

Comment:
You could rewrite this in one `if` as follows:

`if (!(methodOpt is null)) || Binder.TryGetSpecialTypeMember(_compilation, SpecialMember.System_IDisposable__Dispose, syntax, _diagnostics, out methodOpt))`